controllers/overlaynetwork_controller.go

```diff
@@ -60,3 +87,42 @@ func (r *OverlayNetworkReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&selfservicev1.OverlayNetwork{}).
 		Complete(r)
 }
+
+func renderNetAttachDef(overlayNet *selfservicev1.OverlayNetwork) (*netv1.NetworkAttachmentDefinition, error) {
+	const netAttachDefKind = "NetworkAttachmentDefinition"
+	const netAttachDefAPIVer = "v1"
+
+	cniNetConf := map[string]interface{}{
+		"type":             "ovn-k8s-cni-overlay",
+		"name":             overlayNet.GetName(),
+		"netAttachDefName": overlayNet.Namespace + "/" + overlayNet.Name,
+		"topology":         "layer2",
+	}
+	cniNetConfRaw, err := json.Marshal(cniNetConf)
+	if err != nil {
+		return nil, err
+	}
+
+	return &netv1.NetworkAttachmentDefinition{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: netAttachDefAPIVer,
+			Kind:       netAttachDefKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      overlayNet.Name,
+			Namespace: overlayNet.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
```

Comment:
If this repo create it, you might want the controller field to be true ?

What about blockownerDeletion ? which matters in foreground deletion, in case you want the nad to be deleted before the OverlayNetwork, but i guess it is not important in this case

Reply:
I agree, the controller field is missing. And makes sense to keep the default `BlockOwnerDeletion` value which is true.

Reply:
Done

Reply:
I still don't see the controller field.